src/include/units/random.h

```diff
@@ -0,0 +1,453 @@
+    
+    Quantity min() const { return Quantity(Base::min()); }
+    Quantity max() const { return Quantity(Base::max()); }
+};
+
+template<typename Quantity>
+struct student_t_distribution : public std::student_t_distribution<typename Quantity::rep>
+{
+    using Rep = Quantity::rep;
+    using Base = std::student_t_distribution<Rep>;
+    
+    student_t_distribution() : Base() {}
+    student_t_distribution(Rep n) : Base(n) {}
+    
+    template<typename Generator>
+    Quantity operator()(Generator& g) { return Quantity(Base::operator()(g)); }
+    
+    Quantity min() const { return Quantity(Base::min()); }
+    Quantity max() const { return Quantity(Base::max()); }
+};
+
+template<typename Quantity>
+struct discrete_distribution : public std::discrete_distribution<typename Quantity::rep>
+{
+    using Rep = Quantity::rep;
+    using Base = std::discrete_distribution<Rep>;
+    
+    discrete_distribution() : Base() {}
+    
+    template <typename InputIt>
+    discrete_distribution(InputIt first, InputIt last) : Base(first, last) {}
+
+    discrete_distribution(std::initializer_list<double> weights) : Base(weights) {}
+
+    template <typename UnaryOperation>
+    discrete_distribution(std::size_t count, double xmin, double xmax, UnaryOperation unary_op) :
+        Base(count, xmin, xmax, unary_op) {}
+    
+    template<typename Generator>
+    Quantity operator()(Generator& g) { return Quantity(Base::operator()(g)); }
+    
+    Quantity min() const { return Quantity(Base::min()); }
+    Quantity max() const { return Quantity(Base::max()); }
+};
+
+template<typename Quantity>
+class piecewise_constant_distribution : public std::piecewise_constant_distribution<typename Quantity::rep>
+{
+    using Rep = Quantity::rep;
+    using Base = std::piecewise_constant_distribution<Rep>;
+
+    template <typename InputIt>
+    inline static std::vector<Rep> i_qty_to_rep(InputIt first, InputIt last)
+    {
+        std::vector<Rep> intervals_rep;
+        intervals_rep.reserve(static_cast<size_t>(std::distance(first, last)));
+        for (auto itr = first; itr != last; ++itr) { intervals_rep.push_back(itr->count()); }
+        return intervals_rep;
+    }
+    
+    inline static std::vector<Rep> bl_qty_to_rep(std::initializer_list<Quantity>& bl)
+    {
+        std::vector<Rep> bl_rep;
+        bl_rep.reserve(bl.size());
+        for (const Quantity& qty : bl) { bl_rep.push_back(qty.count()); }
+        return bl_rep;
+    }
+
+    template <typename UnaryOperation>
+    inline static std::vector<Rep> fw_bl(std::initializer_list<Quantity>& bl, UnaryOperation fw)
+    {
+        std::vector<Rep> w_bl;
+        w_bl.reserve(bl.size());
+        for (const Quantity& qty : bl) { w_bl.push_back(fw(qty)); }   
+        std::vector<Rep> weights;
+        weights.reserve(bl.size());
+        for (size_t i = 0; i < bl.size() - 1; ++i) { weights.push_back(w_bl[i] + w_bl[i + 1]); }
+        weights.push_back(0);
+        return weights;
+    }
```

Comment:
Please make those non-member functions in `detail` namespace. Besides, do not put `inline` for member functions as they are `inline` already ;-)

Reply:
Alright, noted.